lib/auth/auth_with_roles.go

```diff
@@ -1101,8 +1101,9 @@ func (a *ServerWithRoles) RegisterInventoryControlStream(ics client.UpstreamInve
 }
 
 func (a *ServerWithRoles) GetInventoryStatus(ctx context.Context, req proto.InventoryStatusRequest) (proto.InventoryStatusSummary, error) {
-	// only support builtin roles for now, but we'll eventually want to develop an RBAC syntax for
-	// the inventory APIs once they are more developed.
+	// TODO(fspmarshall): switch this to being scoped to instance:read once we have a sane remote version of
+	// this method. for now we're leaving it as requiring local admin because the returned value is basically
+	// nonsense if you aren't connected locally.
```

Comment:
The linked issue is really solved by enabling instance heartbeats again.  For what they wanted they should have been calling `GetInstances` rather than `GetInventoryStatus`.

I've been waffling on what to do with `GetInventoryStatus`.  Originally it was intended to be used to summarize the ongoing status of an upgrade back when teleport managed upgrades directly.  Now its purpose is a bit ambiguous.

I've added another separate commit to this PR that contains my current best attempt at making it useful for remote users.  Gives a UX something like this:

```
$ tctl inventory status
Total Instances: 2
Services:
  Auth:  1
  Proxy: 1
  Node:  2
Versions:
  v14.0.0-dev: 1
  v13.1.0:     1
Upgraders:
  none: 1
  unit: 1
```

I'm not totally happy with, but the other options for a summary UX that I've tried were even less helpful.